standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -585,10 +586,18 @@ private Path processPathDepthInfo(final PathDepthInfo pd)
       if (currentDepth == partColNames.size()) {
         return currentPath;
       }
-      FileStatus[] fileStatuses = fs.listStatus(currentPath, FileUtils.HIDDEN_FILES_PATH_FILTER);
```

Comment:
Can use `RemoteIterators.filteringRemoteIterator`
there is one wrapper in FileUtils as well.
https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/common/FileUtils.java#L1339-L1342

Reply:
Updated @ayushtkn